RetryOperationHelper/RetryOperationHelper.cs

```diff
@@ -35,13 +35,13 @@ public static class RetryOperationHelper
             {
                 if (attempt > 0 && retryInterval != null)
                 {
-                    await Task.Delay(retryInterval.Value);
+                    await Task.Delay(retryInterval.Value).ConfigureAwait(true);
                 }
 
                 try
                 {
                     //Call the function passed in by the caller. 
-                    return await func().ConfigureAwait(false);
+                    return await func().ConfigureAwait(true);
```

Comment:
The context should be captured so the retry executes under the same context as the first execution.